### PR TITLE
listen to command a and select the active element.

### LINKED ISSUE #5023: Cmd+a in Notebook Outputs Doesn't Do Anything on Mac
On PC and Linux using the keyboard shortcut to select all in a notebook cell's output selects in the output.

On Mac, nothing happens.

### PR DESCRIPTION
**Pull request by MaddyDev**
This PR fixes #5023 

On windows, ctrl-A selects the html element but on Mac it invokes the editors select all action even when there is no editor. So added an event on textCell component to listen to cmd-A key event and prevent default behavior & select the active html element.

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 1]
```diff
@@ -68,6 +68,14 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		this._model.updateActiveCell(undefined);
 	}
 
+	@HostListener('document:keydown.meta.a', ['$event'])
+	onkeydown(e) {
+		// use preventDefault() to avoid invoking the editors select all
+		// select the active .
```
Comment (corivera) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts]: missing a word here

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 2]
```diff
@@ -68,6 +68,14 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		this._model.updateActiveCell(undefined);
 	}
 
+	@HostListener('document:keydown.meta.a', ['$event'])
+	onkeydown(e) {
+		// use preventDefault() to avoid invoking the editors select all
```
Comment (corivera) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts]: missing apostrophe on "editors"

**Comment by Charles-Gagnon** [comment]
Is there anywhere else we do something like this? Do you know why this works on Windows just fine?

**Comment by MaddyDev** [comment]
> Is there anywhere else we do something like this? Do you know why this works on Windows just fine?

Did you mean if we're preventing the default behavior anywhere else? If so yes, in a lot of places (just search for e.preventDefault())
And as for why this works on windows, I don't clearly know either. But I learned that windows and mac handles the events differently, that's probably why both are invoking different flows when I debugged.

**Comment by Charles-Gagnon** [comment]
No, I'm talking about the host listening and doing the selectAll event. I don't see that used anywhere else so I'm just worried we're doing something hacky here when there's a better solution out there (although I'll admit I don't have one right now).

What do you mean by windows "handles events differently"? Do you know if the select all functionality on windows is something that's just default behavior we picked up from the editor?

Did you try doing select all from a normal query editor results output as well to see if the issue repros there?

And another thought - this I assume works fine for the editor cells right? Did you look into why they seem to be fine but this isn't?

Sorry for all the questions - in the end I don't have a huge issue with this but I just want to make sure we're not doing going to hide/ignore other issues by doing something weird.

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 3]
```diff
@@ -68,6 +68,14 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		this._model.updateActiveCell(undefined);
 	}
 
+	@HostListener('document:keydown.meta.a', ['$event'])
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts]: Also - what if the user changes their keybinding for select all? We should either respect the setting the user has for editors or at least make our own.

**Comment by MaddyDev** [comment]
> No, I'm talking about the host listening and doing the selectAll event. I don't see that used anywhere else so I'm just worried we're doing something hacky here when there's a better solution out there (although I'll admit I don't have one right now).
> 
> What do you mean by windows "handles events differently"? Do you know if the select all functionality on windows is something that's just default behavior we picked up from the editor?
> 
> Did you try doing select all from a normal query editor results output as well to see if the issue repros there?
> 
> And another thought - this I assume works fine for the editor cells right? Did you look into why they seem to be fine but this isn't?
> 
> Sorry for all the questions - in the end I don't have a huge issue with this but I just want to make sure we're not doing going to hide/ignore other issues by doing something weird.

We're listening to esc key event to get out of edit mode. It's similar to that. The reason I added this is because mac's behavior is different (which I tried to look up but didn't find anything useful). Unless we find out why that is, I think this the way. 
Query editor and code cell both have editors associated with them and there is a selectAll event bound to ctr/cmd A for editors. So this issue doesn't repro there.

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts (hunk 3 above)
Comment (MaddyDev) [reply src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts]: If they do, it only applies to editors and the changes I made is on an angular component that doesn't have an editor. So probably we can document that somewhere? (Although I think we can agree that copy/paste/selectAll are something that users wouldn't want to change)

**Comment by MaddyDev** [comment]
https://github.com/microsoft/azuredatastudio/issues/8989 to track a bigger issue that arises from this mac behavior.

**Pull request merged**